src/zkp/pisch.rs

```diff
@@ -109,20 +113,21 @@ impl Proof2 for PiSchProof {
         transcript: &mut Transcript,
     ) -> Result<()> {
         // First check Fiat-Shamir challenge consistency
-        Self::fill_transcript(transcript, context, &input, &self.A)?;
+        Self::fill_transcript(transcript, context, &input, &self.commitment)?;
 
         // Verifier samples e in F_q
-        let e = positive_challenge_from_transcript(transcript, input.q)?;
-        if e != self.e {
+        let random_challenge = positive_challenge_from_transcript(transcript, input.q)?;
+        if random_challenge != self.challenge {
```

Comment:
todo: I prefer if this is just called `challenge` rather than `random_challenge` since it's drawn deterministically from the transcript.